src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs

```diff
@@ -454,9 +454,10 @@ private void LbnSelChange()
         if (!_killnextselect && (index == _lastSelected || CheckOnClick))
         {
             CheckState currentValue = CheckedItems.GetCheckedState(index);
-            CheckState newValue = (currentValue != CheckState.Unchecked)
+            CheckState newValue = (currentValue == CheckState.Unchecked)
```

Comment:
I'm not quite following how this works, if `currentValue` is `CheckState.Unchecked`, then `newValue` will also be `CheckState.Unchecked`. Shouldn't it be as before?
```suggestion
            CheckState newValue = (currentValue != CheckState.Unchecked)
```

Reply:
This is just for readability, if it remains the same, the code should be like
`CheckState newValue =`
          `currentValue != CheckState.Unchecked`
          `? currentValue != CheckState.Checked ? CheckState.Indeterminate : CheckState.Checked`
         ` : CheckState.Unchecked;`